File: HA2_4.java
//Accept number from user and check wheather it is even or odd
import java.util.*;

class Number
{
    public boolean ChkNo(int iNo)
    {
        if((iNo %2)==0)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

public class HA2_4 
{
    public static void main(String[] args) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter a number");
        int iValue1 = sobj.nextInt();

        Number nobj= new Number();
        {
            boolean bRet = nobj.ChkNo(iValue1);

            if(bRet == true)
            {
                System.out.println("EVEN");
            }
            else
            {
                System.out.println("ODD");
            }
        }
        sobj.close();
    }    
}

File: HA2_4.ts
import * as readline from 'readline';

function ChkNo(iNo : number)
{
    if((iNo % 2)==0)
    {
        return true
    }
    else
    {
        return false
    }
}

async function main()
{
    const rl = readline.createInterface({
        input:process.stdin,
        output:process.stdout
    })

    let iValue : number = 0;
    let bRet : boolean = false;

    iValue = parseInt(await askQuestion(rl, "Enter a number     ")) ||0;

    bRet = ChkNo(iValue);

    if(bRet == true)
    {
        console.log("EVEN");
    }
    else
    {
        console.log("ODD");
    }

    rl.close();
}

function askQuestion(rl: readline.Interface, question: string): Promise<string>
{
    return new Promise((resolve) =>{
        rl.question(question,(answer : any) =>{
            resolve(answer);
        })
    })
}
main()